### ISSUE TITLE
Hyperparameter tuning for PPO training

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/bucket-brigade#11]
## Goal
Systematically tune hyperparameters to find optimal training configuration for Bucket Brigade.

## Background
Current training uses default hyperparameters that may not be optimal for this multi-agent cooperation task. Key parameters to explore:

### Network Architecture
- Hidden layer size (current: 128)
- Number of layers (current: 2)
- Activation functions

### PPO Parameters
- Learning rate (current: 3e-4)
- Batch size (current: 2048)
- Number of epochs (current: 4)
- Clip epsilon (current: 0.2)
- Entropy coefficient (current: 0.01)
- Value coefficient (current: 0.5)
- GAE lambda (current: 0.95)
- Discount factor gamma (current: 0.99)

## Tasks
- [ ] Create hyperparameter sweep configuration
- [ ] Test different network sizes (64, 128, 256, 512)
- [ ] Test learning rates (1e-5, 3e-5, 1e-4, 3e-4, 1e-3)
- [ ] Test batch sizes (512, 1024, 2048, 4096)
- [ ] Test entropy coefficients (0.001, 0.01, 0.05)
- [ ] Document best performing configurations
- [ ] Update train_simple.py with recommended defaults

### Grid Search
```bash
for lr in 1e-4 3e-4 1e-3; do
  for hidden in 128 256; do
    for batch in 1024 2048; do
      uv run python scripts/train_simple.py \
        --num-steps 100000 \
        --lr $lr \
        --hidden-size $hidden \
        --batch-size $batch \
        --save-path "models/tune_lr${lr}_h${hidden}_b${batch}.pt"
    done
  done
done
```

### Random Search
Consider using tools like:
- Optuna
- Ray Tune
- Weights & Biases sweeps

## Success Criteria
- Identify configuration that improves mean reward by >50%
- Reduce variance by >30%
- Document reproducible hyperparameter settings

## Related
- Issue #10 (longer training runs)
- See TRAINING_GUIDE.md section on "Improve Performance"
